idaes/core/util/scaling.py

```diff
@@ -49,12 +50,36 @@
 from pyomo.core import expr as EXPR
 from pyomo.common.numeric_types import native_types
 from pyomo.core.base.units_container import _PyomoUnit
+from pyomo.common.fileutils import find_library
 
 import idaes.logger as idaeslog
 
 _log = idaeslog.getLogger(__name__)
 
 
+# There appears to be a bug in the ASL which causes terminal failures
+# if you try to create multiple ASL structs with different external
+# functions in the same process. This causes pytest to crash during testing.
+# To avoid this, register all known external functions before we call
+# PyNumero.
+ext_funcs = ["cubic_roots", "general_helmholtz_external", "functions"]
+library_set = set()
+libraries = []
+
+for f in ext_funcs:
+    library = find_library(f)
+    if library not in library_set:
+        library_set.add(library)
+        libraries.append(library)
+
+if "AMPLFUNC" in os.environ:
+    env_str = "\n".join([os.environ["AMPLFUNC"], *libraries])
```

Comment:
I would do something like:
```python
orig_libraries = os.environ["AMPLFUNC"].split("\n")
env_str = "\n".join(orig_libraries + libraries)
```
This is to guard against unexpected behavior if `"\n\n"` appears in `AMPLFUNC`. Hopefully this wouldn't be a problem for ASL, but I have been surprised before...